### DISCUSSION TITLE
How to work with minification?

### DISCUSSION THREAD

**Discussion by QuickScoP3s** [karmaniverous/serify-deserify#27] [Q&A]
I'm using serify for use with Redux, because I wanted a global setup to save luxon's DateTime object to the store.
However, using Vite the build will be minified, thus the name "DateTime" does not work, as `constructor.name` will return something random.

How can I configure serify to correctly identify library objects that don't have the `serifyStaticTypeProperty`?

Many thanks!

**Comment by karmaniverous** [comment]
Hm this is a use case I hadn't thought of.

What if you defined your options object dynamically?

```ts
const customOptions: SerifyOptions<CustomFooTypeMap> = {
  ...defaultOptions
  serifyKey: 42,
  types: {
    ...defaultOptions.types,
    [typeof Foo]: {
      serifier: (value) => value.p,
      deserifier: (value) => new CustomFoo(value)
    }
  }
}
```

... or something like that, don't know offhand if `typeof` output will work as an object key.

I'll think about it, and meanwhile if you come up with a dynamic approach that works, let me know and I'll integrate it.

Or here's a much simpler approach that will get you moving right away: create a wrapper class for the luxon DateTime that DOES have `serifyStaticTypeProperty`.

**Reply by QuickScoP3s** [reply]
I tried the `typeof` solution, but that doesn't seem to work.

My current working solution is this:
```ts
const storeMapping: SerifyOptions<DefaultTypeMap> = {
   ...defaultOptions,
   serifyKey: 'ReduxStore',
   types: {
      ...defaultOptions.types
   }
}

// Dynamically get the DateTime constructor name
const dateTime = DateTime.prototype.constructor.name;
storeMapping.types[dateTime] = {
   serifier: (value: DateTime) => value.toISO()!,
   deserifier: (value: string) => DateTime.fromISO(value)
}

export default storeMapping
```

This isn't very clean, but it works both during debugging and when built using minification.
The 'problem' is that you can't define a `TypeMap` like this of course.
